batch/batch/worker/worker.py

```diff
@@ -408,36 +410,38 @@ def __init__(self, job, name, spec):
         self.overlay_path = None
 
         self.image_config = None
+        self.image_digest = None
         self.rootfs_path = None
         scratch = self.spec['scratch']
         self.container_scratch = f'{scratch}/{self.name}'
         self.container_overlay_path = f'{self.container_scratch}/rootfs_overlay'
         self.config_path = f'{self.container_scratch}/config'
 
+        self.container_name = f'batch-{self.job.batch_id}-job-{self.job.job_id}-{self.name}'
+
         self.netns: Optional[NetworkNamespace] = None
         self.process = None
 
-    async def run(self, worker):
+    async def run(self, worker: 'Worker'):
         try:
 
             async def localize_rootfs():
                 async with worker.rootfs_locks[self.image_ref_str]:
                     last_fetched = worker.rootfs_last_fetched.get(self.image_ref_str)
                     five_minutes_ago = time_msecs() - 5 * 60 * 1000
-                    if last_fetched and last_fetched < five_minutes_ago:
-                        assert self.image_ref_str in image_configs
-                        image_digest = get_image_digest(image_configs[self.image_ref_str])
-                        shutil.rmtree(f'/host/rootfs/{image_digest}')
                     if not last_fetched or last_fetched < five_minutes_ago:
-                        worker.rootfs_last_fetched[self.image_ref_str] = time_msecs()
                         await self.pull_image()
+                        worker.rootfs_last_fetched[self.image_ref_str] = time_msecs()
+                        log.info(f'Added image to cache: {self.image_ref_str}')
```

Comment:
I'm not seeing where the XFS Quota gets set for the container's root file system. What I think I'm envisioning is we pull or use the cached image and get it's total size. We then add XFS quotas for the merged / upper directories (user writable ones) to the remainder of the quota (asserting not less than 0). For example, we have a 1 core job. We've promised the root file system (not including /io) accounts for 5Gi. The image is 2.5Gi. We then change the size of the XFS Quota for the merged/upper directories for the container to 2.5Gi. `/io` is whatever the user requested and how we calculate the quotas and mount `/io` shouldn't change with this PR.